### PR TITLE
elixirscript 0.16.0

### PR DESCRIPTION
**Pull request by ktec**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

_You can erase any parts of this template not applicable to your Pull Request._

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

File: Library/Formula/elixirscript.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def install
+    system "mix", "local.hex", "--force"
+    system "mix", "deps.get"
+    system "mix", "escript.build"
+    bin.install "elixirscript"
+    prefix.install "priv/Elixir.js", "LICENSE"
+  end
+
+  test do
+    system "elixirscript", "-h"
```
Comment (mikemcquaid) [inline Library/Formula/elixirscript.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by ktec** [comment]
@mikemcquaid I wasn't able to follow the examples you cited, nevertheless this change should provide a more substantial assertion.

File: Library/Formula/elixirscript.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    system "mix", "local.hex", "--force"
+    system "mix", "deps.get"
+    system "mix", "escript.build"
+    bin.install "elixirscript"
+    prefix.install "priv/Elixir.js", "LICENSE"
+  end
+
+  test do
+    assert_match /Symbol.for('atom')/,
+                 shell_output("#{bin}/elixirscript \":atom\" -ex")
```
Comment (mikemcquaid) [inline Library/Formula/elixirscript.rb]: Just indent this by two characters. Also, this test is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/43223/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_elixirscript/

**Comment by ktec** [comment]
@mikemcquaid refactored and reformatted, hope that helps.

File: Library/Formula/elixirscript.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    src_path = testpath/"Example.exjs"
+    src_path.write <<-EOS.undent
+      :keith
+    EOS
+
+    out_path = testpath/"dest"
+    system "elixirscript", src_path, "-o", out_path
+
+    assert File.exist?(out_path)
+    assert File.open(out_path/"Elixir.ElixirScript.Temp.js").grep(/keith/).any?
```
Comment (mikemcquaid) [inline Library/Formula/elixirscript.rb]: `assert_match("keith", (out_path/"Elixir.ElixirScript.Temp.js").read)` will probably produce results that are easier to debug on a failure.

**Comment by mikemcquaid** [comment]
@ktec One last note then should be good to go.

**Comment by ktec** [comment]
Thanks for all the feedback @mikemcquaid :+1:

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**